prompting/tools/datasets/base.py

```diff
@@ -117,10 +117,10 @@ def get(self, params: dict):
             "extra": {},
         }
 
-    def random(self, selector: Selector = None):
+    def random(self, selector: Optional[Selector] = None)->Dict[str,Any]:
```

Comment:
We are moving to python `3.10`, and starting from `3.9` there is no need to use `typing.Dict` for type hints. Same for all built-in datastructures

```suggestion
    def random(self, selector: Optional[Selector] = None) -> dict[str,Any]:
```